### ISSUE TITLE
Not OK: perl 5.00504 on i686-linux 2.4.18-bf2.4

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#7141]
Migrated from [rt.perl.org#27188](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=27188) (status was 'resolved')

Searchable as RT27188$

**Comment by p5pRT** [comment]
### From @abigail



'make test' fails&#8203;:

lib/db\-btree\.\.\.\.\.\.\.Can't locate DB\_File\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: \.\./lib\) at lib/db\-btree\.t line 21\.
BEGIN failed\-\-compilation aborted at lib/db\-btree\.t line 21\.
Scalars leaked&#8203;: 1
FAILED at test 0

<details><summary>Perl Info</summary>

```


Site configuration information for perl 5.00504:

Configured by abigail at Thu Feb 19 23:54:08 CET 2004.

Summary of my perl5 (5.0 patchlevel 5 subversion 4) configuration:
  Platform:
    osname=linux, osvers=2.4.18-bf2.4, archname=i686-linux
    uname='linux alexandra 2.4.18-bf2.4 #1 son apr 14 09:53:28 cest 2002 i686 unknown '
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef useperlio=undef d_sfio=undef
  Compiler:
    cc='gcc', optimize='-g', gccversion=3.0.4
    cppflags='-Dbool=char -DHAS_BOOL -DDEBUGGING -I/usr/local/include'
    ccflags ='-Dbool=char -DHAS_BOOL -DDEBUGGING -I/usr/local/include'
    stdchar='char', d_stdstdio=undef, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -ldl -lm -lc -lcrypt
    libc=, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl 5.00504:
    lib
    /home/abigail/Perl
    /opt/perl/5.005_04-RC2/lib/5.00504/i686-linux
    /opt/perl/5.005_04-RC2/lib/5.00504
    /opt/perl/5.005_04-RC2/lib/site_perl/5.005/i686-linux
    /opt/perl/5.005_04-RC2/lib/site_perl/5.005
    .


Environment for perl 5.00504:
    HOME=/home/abigail
    LANG=C
    LANGUAGE (unset)
    LD_LIBRARY_PATH=/home/abigail/Lib:/usr/local/lib:/usr/lib:/lib:/usr/X11R6/lib
    LOGDIR (unset)
    PATH=/home/abigail/Bin:/opt/perl/bin:/usr/local/bin:/usr/local/X11/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/X11R6/bin:/usr/games:/usr/share/texmf/bin:/opt/Acrobat/bin:/opt/java/blackdown/j2sdk1.3.1/bin:/usr/local/games/bin
    PERL5LIB=/home/abigail/Perl
    PERLDIR=/opt/perl
    PERL_BADLANG (unset)
    SHELL=/bin/bash

----- End forwarded message -----

```
</details>

**Comment by p5pRT** [comment]
### From @mhx

On 2004\-02\-28\, at 00&#8203;:44&#8203;:43 \+0100\, Abigail wrote&#8203;:

> 'make test' fails&#8203;:
> 
> lib/db\-btree\.\.\.\.\.\.\.Can't locate DB\_File\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;: \.\./lib\) at lib/db\-btree\.t line 21\.
> BEGIN failed\-\-compilation aborted at lib/db\-btree\.t line 21\.
> Scalars leaked&#8203;: 1
> FAILED at test 0

This should be fixed by the patch I suggested recently in&#8203;:

&nbsp;  http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/88793

Leon\, is this going to be applied?

Marcus

\-\- 
O'Brian's Law&#8203;:
&nbsp;    Everything is always done for the wrong reasons\.

**Comment by p5pRT** [comment]
### From @smpeters

> \[mhx\-perl@&#8203;gmx\.net \- Fri Feb 27 18&#8203;:04&#8203;:33 2004\]&#8203;:
> 
> On 2004\-02\-28\, at 00&#8203;:44&#8203;:43 \+0100\, Abigail wrote&#8203;:
> 
> > 'make test' fails&#8203;:
> >
> > lib/db\-btree\.\.\.\.\.\.\.Can't locate DB\_File\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;:
> \.\./lib\) at lib/db\-btree\.t line 21\.
> > BEGIN failed\-\-compilation aborted at lib/db\-btree\.t line 21\.
> > Scalars leaked&#8203;: 1
> > FAILED at test 0
> 
> This should be fixed by the patch I suggested recently in&#8203;:
> 
>   http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/88793
> 
> Leon\, is this going to be applied?
> 

Two issues in this ticket\.  First\, Marcus's patch in the email shown
just above has not been applied to the 5\.00504 branch\.  Second\, the
perbug email addresses for 5\.00504 branch are incorrect\, but are fixed
in the patch below\.

<details><summary>Inline Patch</summary>

```diff
--- utils/perlbug.PL.old        2002-07-19 18:48:57.000000000 -0500
+++ utils/perlbug.PL    2005-09-25 22:31:24.000000000 -0500
@@ -163,10 +163,10 @@
     # -------- Configuration ---------

     # perlbug address
-    $perlbug = 'perlbug@perl.com';
+    $perlbug = 'perlbug@perl.org';

     # Test address
-    $testaddress = 'perlbug-test@perl.com';
+    $testaddress = 'perlbug-test@perl.org';

     # Target address
     $address = $::opt_a || ($::opt_t ? $testaddress : $perlbug);
@@ -870,7 +870,7 @@

 If you are unable to run B<perlbug> (most likely because you don't have
 a working setup to send mail that perlbug recognizes), you may have to
-compose your own report, and email it to B<perlbug@perl.com>.  You might
+compose your own report, and email it to B<perlbug@perl.org>.  You might
 find the B<-d> option useful to get summary information in that case.

 In any case, when reporting a bug, please make sure you have run through
@@ -948,7 +948,7 @@
 (C<diff> is being maintained by the GNU folks as part of the B<diffutils>
 package, so you should be able to get it from any of the GNU software
 repositories).  If you do submit a patch, the cool-dude counter at
-perlbug@perl.com will register you as a savior of the world.  Your
+perlbug@perl.org will register you as a savior of the world.  Your
 patch may be returned with requests for changes, or requests for more
 detailed explanations about your fix.

@@ -968,7 +968,7 @@
 crucial information about your version of perl.  If C<perlbug> is unable
 to mail your report after you have typed it in, you may have to compose
 the message yourself, add the output produced by C<perlbug -d> and email
-it to B<perlbug@perl.com>.  If, for some reason, you cannot run
+it to B<perlbug@perl.org>.  If, for some reason, you cannot run
 C<perlbug> at all on your system, be sure to include the entire output
 produced by running C<perl -V> (note the uppercase V).

@@ -995,7 +995,7 @@

 =item B<-a>

-Address to send the report to.  Defaults to `perlbug@perl.com'.
+Address to send the report to.  Defaults to `perlbug@perl.org'.

 =item B<-b>

@@ -1080,7 +1080,7 @@

 =item B<-t>

-Test mode.  The target address defaults to `perlbug-test@perl.com'.
+Test mode.  The target address defaults to `perlbug-test@perl.org'.

 =item B<-v>
```
</details>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @smpeters

On Thu\, Sep 29\, 2005 at 09&#8203;:15&#8203;:05AM \+0100\, Paul Marquess wrote&#8203;:
> From&#8203;: Steve Peters via RT \[mailto&#8203;:perlbug\-followup@&#8203;perl\.org\]
>  
> > > \[mhx\-perl@&#8203;gmx\.net \- Fri Feb 27 18&#8203;:04&#8203;:33 2004\]&#8203;:
> > >
> > > On 2004\-02\-28\, at 00&#8203;:44&#8203;:43 \+0100\, Abigail wrote&#8203;:
> > >
> > > > 'make test' fails&#8203;:
> > > >
> > > > lib/db\-btree\.\.\.\.\.\.\.Can't locate DB\_File\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;:
> > > \.\./lib\) at lib/db\-btree\.t line 21\.
> > > > BEGIN failed\-\-compilation aborted at lib/db\-btree\.t line 21\.
> > > > Scalars leaked&#8203;: 1
> > > > FAILED at test 0
> > >
> > > This should be fixed by the patch I suggested recently in&#8203;:
> > >
> > >   http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/88793
> > >
> > > Leon\, is this going to be applied?
> 
> I can't answer for the 5\.0050x codeline\, but I can certainly apply the
> change to the CPAN and 5\.9\.3 versions of DB\_File if you like\.
> 

That'd be great if you could\.  Thanks\!

Steve Peters
steve@&#8203;fisharerojo\.org

**Comment by p5pRT** [comment]
p5p@spam.wizbit.be - Status changed from 'open' to 'resolved'
